### ISSUE TITLE
TypeError: nextytext.toDelta is not a function or its return value is not iterable

### ISSUE THREAD

**Issue by danny-hunt** [yjs/y-prosemirror#163]
Crash caused by upgrading to 1.2.11

**Checklist**

* [X] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [X] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
`TypeError: nextytext.toDelta is not a function or its return value is not iterable` shows on performing a specific action in the editor.
This caught error escalates to a crash on other actions (eg changing page)

I added some console logging in sync-plugin.js on my machine (logging out the type & nextytext):
<img width="530" alt="image" src="https://github.com/user-attachments/assets/cc29e8f7-477e-4ae6-828a-78407d6d978f">

And get these logs - showing an instance where nextytext does not have a toDelta function.
<img width="858" alt="image" src="https://github.com/user-attachments/assets/13c0f668-4132-4078-856a-878540455cbd">

The command function here is the one that triggers the error for me. There must be more than one user collaborating at the same time for the bug to occur
<img width="371" alt="image" src="https://github.com/user-attachments/assets/ca5741c2-beb6-463e-954b-8b268386e6f6">

I've patched the error locally via
<img width="557" alt="image" src="https://github.com/user-attachments/assets/3d079d85-1968-4095-a0d9-1518b8be8f5b">

**To Reproduce**
Apologies that I don't have a simple example case for this
1. Two users in collaboration
2. Open console
3. Perform action that results in a nextytext that does not have a toDelta function (eg a YXmlElement)

**Expected behavior**
Would not expect a strict requirement for nextytext to have a toDelta function

**Screenshots**
Added above

**Environment Information**
 -  node v22.4.0
- chrome 127.0.6533.99 (Official Build) (arm64) (but it occurs across many setups
- yjs 13.6.18
- y-indexeddb 9.0.12
- y-partykit 0.0.31
- y-protocols 1.0.6
- @syncedstore/yjs-reactive-bindings@0.6.0

**Additional context**

* [ ] I'm a sponsor 💖
* [ ] This issue is a blocker for my project.

**Comment by manstie** [comment]
I am also getting this issue when trying to copy an editor's contents to another using this:
```
xml2.insert(
    0,
    xml1
	.toArray()
	.filter((item) => !(item instanceof XmlHook))
	.map((item) => item.clone()),
);
```

**Comment by dmonad** [comment]
Thank you for the bug report. The issue should be fixed in the latest release (1.2.12). Thanks to @manstie  we have a bugfix.  Please let me know if there is still an issue.
